services/learn-card-network/brain-service/src/cache/invites.ts

```diff
@@ -4,29 +4,75 @@ export const getInviteCacheKey = (profileId: string, challenge: string): string
     `inviteChallenge:${profileId}:${challenge}`;
 
 export const VALID = 'valid';
+export const NEVER_EXPIRE = 'never';
 
 export const isInviteValidForProfile = async (
     profileId: string,
     challenge: string,
-): Promise<typeof VALID | null | undefined> => {
+): Promise<boolean> => {
     const result = await cache.get(getInviteCacheKey(profileId, challenge));
+    console.log(`isInviteValidForProfile - result: ${result}`);
+
+    // If result is null or undefined, the invite has expired or doesn't exist
+    if (!result) {
+        console.log('isInviteValidForProfile - invite not found or expired');
+        return false;
+    }
 
-    return result === VALID ? result : undefined;
+    const isValid = result === VALID || result === NEVER_EXPIRE;
+    console.log(`isInviteValidForProfile - isValid: ${isValid}`);
+    return isValid;
 };
 
 export const isInviteAlreadySetForProfile = async (
     profileId: string,
     challenge: string,
 ): Promise<boolean> => {
+    console.log(`isInviteAlreadySetForProfile: Checking for profileId: ${profileId}, challenge: ${challenge}`);
     const result = await cache.get(getInviteCacheKey(profileId, challenge));
+    console.log(`isInviteAlreadySetForProfile: Result from cache: ${result}`);
 
     return Boolean(result);
 };
 
-export const setValidInviteForProfile = async (profileId: string, challenge: string, expiration?: number) => {
-    return cache.set(getInviteCacheKey(profileId, challenge), VALID, expiration);
+export const setValidInviteForProfile = async (
+    profileId: string,
+    challenge: string,
+    expiration: number | null
+) => {
+    if (expiration === null) {
+        return cache.set(getInviteCacheKey(profileId, challenge), NEVER_EXPIRE);
```

Comment:
```suggestion
        return cache.set(getInviteCacheKey(profileId, challenge), NEVER_EXPIRE, 0);
```

This is maybe not super well documented rn, but omitting that argument actually sets the TTL to 1 hour! If you set it to 0 (or any falsy value), it will actually be set to not have a TTL!